src/main/java/ca/uhn/fhir/jpa/starter/JpaRestfulServer.java

```diff
@@ -85,11 +95,46 @@ protected void initialize() throws ServletException {
     }
 
     setFhirContext(appCtx.getBean(FhirContext.class));
-    registerProvider(new MessagePlainProvider(appCtx.getBean(DaoRegistry.class), this.getFhirContext()));
-
+    
+    ProcessMessageProvider pmp = new ProcessMessageProvider(this.getFhirContext(), (reqBundle, reqHeader) -> {
+    	DaoRegistry daoRegistry = appCtx.getBean(DaoRegistry.class);
+    	
+        IFhirResourceDao<Bundle> bundleDao = daoRegistry.getResourceDao(ResourceType.Bundle.name());
+        IFhirResourceDao<MessageHeader> messageDao = daoRegistry.getResourceDao(ResourceType.MessageHeader.name());
+        bundleDao.create(reqBundle);
+        messageDao.create(reqHeader);
+    	
+        // NOTE: this line is the reason the provider doesn't do this itself
+        // -- it doesn't know its own address (HapiProperties is JPA server only)
```

Comment:
The main thing was I didn't want to depend on anything that might not always be present. So for example I didn't want to require an environment variable because some HAPI servers don't use those, and I didn't want to assume hapi.properties in a specific location either. And really, we don't actually know what the operation is even supposed to do, so we have to make that configurable anyway. So at best we could move a little bit of the response logic.